samza-operator/src/main/java/org/apache/samza/operators/impl/NoOpOperatorImpl.java

```diff
@@ -24,30 +24,13 @@
 
 
 /**
- * Wrapper class to be used by {@link OperatorImpl}
- *
- * @param <M>  Type of input stream {@link Message}
+ * A no-op operator implementation that simply forwards the incoming message to all of its subscribers.
```

Comment:
I would prefer to name it RootOperatorImpl<> as you suggest, since so far, that's the only usage context and it would be much clear that what we use it for. When more general use cases come, we can easily update it since it is internal implementation. I felt that it might be too early to optimize for general cases since we don't have one yet.